embedding.py

```diff
@@ -0,0 +1,47 @@
+import argparse
+import os
+import time
+
+import numpy as np
+import pandas as pd
+from sentence_transformers import SentenceTransformer
+
+
+def encode(sentences: np.ndarray, model: str = "all-MiniLM-L6-v2") -> None:
+    """Generates the embeddings of sentences using a
+    sentence-transformers model."""
+
+    model = SentenceTransformer("sentence-transformers/" + model)
```

Comment:
Having this here will reload the model every time you call `encode`. You can cache the loaded model instead and save quite some time.

e.g.
```python
from functools import cache

@cache
def get_model(model_name: str = "all-MiniLM-L6-v2"):
    return SentenceTransformer("sentence-transformers/" + model_name)

# inside encode
model = get_model()
```

Also, because we _need_ to use the same model for the cosine similarity to make any sense, I'd say you can remove the `model_name` argument completely

Reply:
Great! Done!